core/src/main/scala/scala/pickling/internal/RuntimePicklerRegistry.scala

```diff
@@ -0,0 +1,61 @@
+package scala.pickling
+package internal
+
+import scala.pickling.runtime.CustomRuntime
+import scala.pickling.spi.PicklerRegistry
+
+/** Helper class to register picklers and unpicklers that
+  * handle special/crazy classes for runtime generation.
+  *
+  * Related to [[scala.pickling.runtime.CustomRuntime]].
+  * Note: Currently this only handles Tuple2s.
+  */
+trait RuntimePicklerRegistry extends CustomRuntime {
```

Comment:
Just an FYI -

I think giving `FastTypeTag.makeRaw(clazz)` "Special powers" to handle specialized instances might actually remove the need for this entire hackery.

So, if `makeRaw` can take a specialized class but dive into the unspecialized version for generating a runtime type tag, we could actually handle a LOT more cases than just Tuple's and most/all of this code disappears.